### PR TITLE
Return early when parent-related record isn't found

### PR DESCRIPTION
**Pull request by MrPetovan**
Address:
- https://github.com/friendica/friendica/issues/10474#issuecomment-927118628
- https://github.com/friendica/friendica/issues/10474#issuecomment-925263894

Not sure if returning early was the correct move in either case, but I couldn't guess the correct values to use instead.

### CODE REVIEW

File: src/Model/Mail.php [hunk 1]
```diff
@@ -76,6 +76,10 @@ public static function insert($msg, $notifiction = true)
 
 		if ($msg['reply']) {
 			$reply = DBA::selectFirst('mail', ['uri', 'uri-id'], ['parent-uri' => $msg['parent-uri'], 'reply' => false]);
+			if (!DBA::isResult($reply)) {
```
Comment (annando) [inline src/Model/Mail.php]: I would prefer a workaround where the mail is kept. Mails are really important, so we should try to keep them.

File: src/Protocol/OStatus.php [hunk 2]
```diff
@@ -1930,6 +1930,10 @@ private static function entryFooter(DOMDocument $doc, $entry, array $item, array
 
 		if ($item['gravity'] != GRAVITY_PARENT) {
 			$parent = Post::selectFirst(['guid', 'author-link', 'owner-link'], ['id' => $item['parent']]);
+			if (!DBA::isResult($parent)) {
```
Comment (annando) [inline src/Protocol/OStatus.php]: The `parent` field is no "real" field but is is build in the view by the `id` of the parent post. This means that it has to exist. So when this query here fails then there has to be some underlying problem, possibly with the database configuration. I wouldn't want to handle it this way.

File: src/Model/Mail.php (hunk 1 above)
Comment (MrPetovan) [reply src/Model/Mail.php]: I'll use the default case then.

**Comment by MrPetovan** [comment]
Reworked the fixes, thank you for the input @annando

**Pull request merged**